frappe/desk/form/load.py

```diff
@@ -31,11 +31,17 @@ def getdoc(doctype, name, user=None):
 	if not name:
 		name = doctype
 
-	if not frappe.db.exists(doctype, name):
+	if not frappe.get_meta(doctype).get("is_virtual") and not frappe.db.exists(doctype, name):
 		return []
 
 	try:
-		doc = frappe.get_doc(doctype, name)
+
+		if frappe.get_meta(doctype).get("is_virtual"):
+			doc = frappe.get_doc(doctype).get_value(doctype, name)
```

Comment:
Shouldn't `get_doc` return a Document object, even for virtual DocTypes 🤔

Reply:
That's part of get_value function, at least is what I did on my local implementation of a Virtual Table :thinking: If you have and idea I can check it :smile:

Reply:
Hi @etobella Have you make Virtual Doctype working on your local environment? What database or datasource are you using? What Frappe version are you using for? No more questions 😄

Reply:
Hi! Yes I did it!!! I used a mongodb as datasource. I used `develop`. Still some issues, but I am solving them one by one :smile:   @git-avc